microsoft/testsuites/dpdk/dpdksuite.py

```diff
@@ -0,0 +1,221 @@
+            self._execute_expect_zero(node, "mv /usr/bin/meson /usr/bin/meson.bak")
+            self._execute_expect_zero(node, "ln -s /usr/local/bin/meson /usr/bin/meson")
+            self._execute_expect_zero(node, "pip3 install --upgrade ninja")
+
+        elif isinstance(node.os, Redhat) or isinstance(node.os, CentOs):
+            self._execute_expect_zero(
+                node, "yum update -y --disablerepo='*' --enablerepo='*microsoft*'"
+            )
+            node.os.install_packages(
+                ["groupinstall", "'Infiniband Support'"], signed=False
+            )  # todo gross hack to support groupinstall
+            for package in self._redhat_packages:
+                node.os.install_packages(package)
+            result = node.execute(
+                "dracut --add-drivers 'mlx4_en mlx4_ib mlx5_ib' -f"
+            )  # add mellanox drivers
+            self.log.debug("\n".join([result.stdout, result.stderr]))
+            self._execute_expect_zero(node, "systemctl enable rdma")
+            self._execute_expect_zero(node, "pip3 install --upgrade meson")
+            self._execute_expect_zero(node, "ln -s /usr/local/bin/meson /usr/bin/meson")
+
+            self._execute_expect_zero(
+                node,
+                f"wget {self._ninja_url}",
+            )
+            self._execute_expect_zero(
+                node, "unzip ninja-linux.zip && mv ninja /usr/bin/ninja"
+            )
+            self._execute_expect_zero(node, "pip3 install --upgrade pyelftools")
+
+    def _execute_expect_zero_with_path(
+        self, node: Node, cmd: str, path: PurePath
+    ) -> str:
+        result = node.execute(cmd, sudo=True, cwd=path, shell=True)
+        result.assert_exit_code()
+        self.log.info(result.stdout)  # TODO: debug
+
+        return result.stdout
+
+    def _execute_expect_zero(self, node: Node, cmd: str) -> str:
+        return self._execute_expect_zero_with_path(node, cmd, node.working_path)
+
+    def _hugepages_init(self, node: Node) -> None:
+        self._execute_expect_zero(node, "mkdir -p /mnt/huge")
+        self._execute_expect_zero(node, "mkdir -p /mnt/huge-1G")
+        self._execute_expect_zero(node, "mount -t hugetlbfs nodev /mnt/huge")
+        self._execute_expect_zero(
+            node, "mount -t hugetlbfs nodev /mnt/huge-1G -o 'pagesize=1G'"
+        )
+
+    def _hugepages_enable(self, node: Node) -> None:
+        self._execute_expect_zero(
+            node,
+            "echo 4096 > /sys/devices/system/node/node0/hugepages/hugepages-2048kB/nr_hugepages",
+        )
+        self._execute_expect_zero(
+            node,
+            "echo 1 > /sys/devices/system/node/node0/hugepages/hugepages-1048576kB/nr_hugepages",
+        )
+        result = node.execute("grep -i huge /proc/meminfo && ls /mnt/", shell=True)
+        self.log.info(f"hugepages status \n{result.stdout}")
+
+    def _install_dpdk(self, node: Node) -> None:
+        git_tool = node.tools[Git]
```

Comment:
Again, packaged dpdk by distros not enough here? It's a default config build, apparently, so... Might as well be there equally for free?

Reply:
I agree and would never do this but most people aren't using the packaged versions apparently. The test also needs some flexibility to be able to validate new release candidates etc. This code will get tweaked to allow for any tagged release version to be installed and tested, so this section isn't finalized yet.